R/msaBayesianGaugeRR.R

```diff
@@ -0,0 +1,2341 @@
+                                            progress = FALSE))
+
+    if(isTryError(bfFit)) {
+      jaspResults[["modelComparison"]][["object"]] <- bfFit
+      return()
+    }
+
+    bfDf <- as.data.frame(bfFit)
+    full <- parts
+    bfFullNull <- bfDf$bf
+
+  } else {
+    formula <- as.formula(paste(measurements, "~", parts, "*", operators))
+
+    # run general comparison for all potential models
+    bfFit <- try(BayesFactor::generalTestBF(formula, data = dataset,
+                                            whichRandom = c(operators, parts),
+                                            rscaleRandom = options$rscalePrior,
+                                            progress = FALSE))
+    if(isTryError(bfFit)) {
+      jaspResults[["modelComparison"]][["object"]] <- bfFit
+      return()
+    }
+    bfDf <- as.data.frame(bfFit)
+
+    # extract full model and model with only main effects
+    main <- paste(parts, "+", operators)
+    full <- paste0(parts, " + ", operators, " + ", parts, ":", operators)
+    bfDf <- bfDf[c(main, full), ]
+
+    bfFullNull <- bfDf[full, ]$bf
+  }
+
+  # dropping unnecessary columns
+  bfDf <- bfDf[, !colnames(bfDf) %in% c("time", "code")]
+
+  # obtain BF comparing full model to other models
+  bfDf$bf <- bfFullNull / bfDf$bf
+
+  # add null model
+  bfDf["Null model", ] <- c(bfFullNull,
+                            bfDf[full, ]$error)
+
+  bfDf[full, ]$error <- ""
+
+  # add model names & change colnames
+  colnames(bfDf) <- c("comparisonBF", "error")
+  bfDf$modelName <- rownames(bfDf)
+  if(!options$type3) {
+    bfDf$modelName <- jaspBase::gsubInteractionSymbol(bfDf$modelName)
+  }
+
+  bfDf <- bfDf[order(-bfDf$comparisonBF), ]
+
+  jaspResults[["modelComparison"]][["object"]] <- bfDf
+
+  return()
+}
+
+#### Plots
+.createContourPlot <- function(jaspResults, parts, operators, measurements, dataset, options) {
+  if(!is.null(jaspResults[["contourPlot"]])) {
+    return()
+  }
+
+  contourPlot <- createJaspContainer(title = gettext("Contour Plot"))
+  contourPlot$position <- 6
+  contourPlot$dependOn(c(.varCompTableDependencies(),
+                         "studyVarianceMultiplierType", "studyVarianceMultiplierValue",
+                         "contourPlot", "contourUSL", "contourLSL"))
+
+  jaspResults[["contourPlot"]] <- contourPlot
+
+  tempPlot <- createJaspPlot(width = 600, height = 600)
+  tempPlot$position <- 2
+
+  samplesMat <- .arrayToMat(jaspResults[["MCMCsamples"]][["object"]])
+  excludeInter <- .evalInter(jaspResults, parts, operators, options)
+  compDf <-.getComponentsFromSamples(jaspResults, parts, operators, options, excludeInter)
+
+  # obtain necessary data
+  contourDf <- compDf[, c("total", "part")]
+  mu <- mean(dataset[[measurements]])
+
+  # data frame for plotting
+  meanEllipse = TRUE
+  plotDf <- .getEllipses(contourDf, mu, meanEllipse = meanEllipse, options = options)
+
+  if(isTryError(plotDf)) {
+    errorMsg <- paste("Failed to calculate contour:", .rmNewLine(plotDf))
```

Comment:
```suggestion
    errorMsg <- gettextf("Failed to calculate contour: %s", .rmNewLine(plotDf))
```
Not sure if you need `%s`, that depends on the output of `.rmNewLine`, but if it's `character` it should be.